src/Models/EventIndex.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace TruckersMP\APIClient\Models;
+
+use TruckersMP\APIClient\Collections\EventCollection;
+
+class EventIndex
+{
+    /**
+     * The featured events.
+     *
+     * @var EventCollection|null
+     */
+    protected $featured;
+
+    /**
+     * Today's events.
+     *
+     * @var EventCollection|null
+     */
+    protected $today;
+
+    /**
+     * The upcoming events.
+     *
+     * @var EventCollection|null
+     */
+    protected $upcoming;
+
+    /**
+     * Create a new EventIndex instance.
+     *
+     * @param  array  $response
+     * @return void
+     */
+    public function __construct(array $response)
+    {
+        if (array_key_exists('featured', $response)) {
+            $this->featured = new EventCollection($response['featured']);
+        }
+
+        if (array_key_exists('today', $response)) {
+            $this->today = new EventCollection($response['today']);
+        }
+
+        if (array_key_exists('upcoming', $response)) {
+            $this->upcoming = new EventCollection($response['upcoming']);
+        }
```

Comment:
I prefer empty collections over making them `null`. What do you think?

Reply:
Returning empty collections indeed seems better concerning code consistency, since that's also done in the `CompanyIndex`. Thanks!

Reply:
Actually, now that I take a second look at why I've implemented it like this, there was a reason behind it.

The API itself doesn't return those array keys if they have no items. For example, see the event index response of the testing website: https://api.truckersmp.network/v2/events, you will see that the `today` key does not exist, since there are no events for this day there.

Do you want me to refactor this code so that, if the key doesn't exist, it returns an empty collection? Or leave it like this 🙂

Reply:
In my opinion, it should just be an empty collection as in this case, it really has no items.
It then makes it easier to work with for developers as well.